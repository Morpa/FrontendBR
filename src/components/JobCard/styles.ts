import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    height: auto;
    border-left: 1rem solid ${theme.colors.primary};
    border-radius: ${theme.border.radius};
    background-color: ${theme.colors.darkGray};
    color: ${theme.colors.white};
    padding: ${theme.spacings.xsmall};
    cursor: pointer;
    transition: 200ms;

    &:hover {
      transform: scale(1.02);
    }
  `}
`

export const Title = styled.h1`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    text-align: justify;
  `}
`
export const LabelsWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: flex-start;
    flex-wrap: wrap;
    width: 100%;
    margin: ${theme.spacings.xxsmall};
  `}
`
export const Label = styled.span`
  ${({ theme }) => css`
    display: flex;
    justify-content: flex-start;
    align-items: center;
    width: auto;
    height: ${theme.spacings.small};
    border-radius: ${theme.border.radius};
    padding: ${theme.spacings.xxsmall};
    margin: 0.4rem;
    font-size: ${theme.font.sizes.xsmall};
    font-weight: ${theme.font.bold};

    > svg {
      cursor: pointer;
    }
  `}
`
export const Publication = styled.span`
  ${({ theme }) => css`
    color: ${theme.colors.lightBg};
    margin-top: ${theme.spacings.xxsmall};
    font-size: ${theme.font.sizes.xsmall};

    ${media.lessThan('medium')`
      width: 100%;
    `}
  `}
`
