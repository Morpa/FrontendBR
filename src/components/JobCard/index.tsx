import formatColor from 'utils/formatColor'
import formatDate from 'utils/formatDate'

import * as S from './styles'

type Labels = {
  name: string
  color: string
}

export type JobCardProps = {
  title: string
  labels: Labels[] | null
  created_at: string
  html_url: string
}

const JobCard = ({ title, labels, created_at, html_url }: JobCardProps) => {
  return (
    <S.Wrapper
      onClick={() => {
        window.open(html_url, '_blank')
      }}
    >
      <S.Title>{title}</S.Title>
      <S.LabelsWrapper>
        {labels &&
          labels.map((label) => (
            <S.Label
              key={label.name}
              style={{
                backgroundColor: `#${label.color}`,
                color: `${formatColor(label.color)}`
              }}
            >
              {label.name}
            </S.Label>
          ))}
      </S.LabelsWrapper>
      <S.Publication>Publicado em: {formatDate(created_at)}</S.Publication>
    </S.Wrapper>
  )
}

export default JobCard
