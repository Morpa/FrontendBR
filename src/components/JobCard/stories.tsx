import { Story, Meta } from '@storybook/react/types-6-0'
import JobCard, { JobCardProps } from '.'

export default {
  title: 'JobCard',
  component: JobCard,
  args: {
    title: '[CURITIBA-PR] Web Designer @ Grupo A Educação',
    labels: [
      { name: 'Pleno', color: '6c46ea' },
      { name: 'CLT', color: '7fe266' }
    ],
    created_at: '2021-02-15T19:52:17Z',
    html_url: 'https://github.com/frontendbr/vagas/issues/4297'
  }
} as Meta

export const Default: Story<JobCardProps> = (args) => (
  <div style={{ width: '34rem', height: 'auto' }}>
    <JobCard {...args} />
  </div>
)
