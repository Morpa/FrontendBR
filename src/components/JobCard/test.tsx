import { screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { renderWithTheme } from 'utils/tests/helpers'

import JobCard from '.'

const props = {
  title: '[CURITIBA-PR] Web Designer @ Grupo A Educação',
  labels: [
    { name: 'Pleno', color: '6c46ea' },
    { name: 'CLT', color: '7fe266' }
  ],
  created_at: '15-02-2021',
  html_url: 'https://github.com/frontendbr/vagas/issues/4297'
}

describe('<JobCard />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<JobCard {...props} />)

    expect(
      screen.getByRole('heading', { name: props.title })
    ).toBeInTheDocument()

    expect(screen.getByText(props.labels[0].name)).toBeInTheDocument()

    expect(screen.getByText(props.labels[0].name)).toHaveStyle({
      backgroundColor: '#6c46ea'
    })

    expect(screen.getByText('Publicado em: 2021/02/15')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render div as link', () => {
    global.open = jest.fn()

    renderWithTheme(<JobCard {...props} />)

    userEvent.click(screen.getByRole('heading', { name: props.title }))

    expect(global.open).toBeCalled()
  })
})
