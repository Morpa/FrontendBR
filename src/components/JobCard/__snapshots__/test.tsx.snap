// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobCard /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  height: auto;
  border-left: 1rem solid #F231A5;
  border-radius: 0.4rem;
  background-color: #2E2F42;
  color: #FAFAFA;
  padding: 1.6rem;
  cursor: pointer;
  -webkit-transition: 200ms;
  transition: 200ms;
}

.c0:hover {
  -webkit-transform: scale(1.02);
  -ms-transform: scale(1.02);
  transform: scale(1.02);
}

.c1 {
  font-size: 1.4rem;
  text-align: justify;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  margin: 0.8rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: auto;
  height: 2.4rem;
  border-radius: 0.4rem;
  padding: 0.8rem;
  margin: 0.4rem;
  font-size: 1.2rem;
  font-weight: 600;
}

.c3 > svg {
  cursor: pointer;
}

.c4 {
  color: #F2F2F2;
  margin-top: 0.8rem;
  font-size: 1.2rem;
}

@media (max-width:768px) {
  .c4 {
    width: 100%;
  }
}

<div
  class="c0"
>
  <h1
    class="c1"
  >
    [CURITIBA-PR] Web Designer @ Grupo A Educação
  </h1>
  <div
    class="c2"
  >
    <span
      class="c3"
      style="background-color: rgb(108, 70, 234); color: rgb(250, 250, 250);"
    >
      Pleno
    </span>
    <span
      class="c3"
      style="background-color: rgb(127, 226, 102); color: rgb(3, 5, 23);"
    >
      CLT
    </span>
  </div>
  <span
    class="c4"
  >
    Publicado em: 
    2021/02/15
  </span>
</div>
`;
